fix(docs-infra): size footer links to appropriate size for seo and small css fixes

ajitsinghkaler opened the pull request:
Footer links did not have enough space between them, so lighouse was reporting that tap targets are not appropriately sized added the required 8px space between links

Fixes #34901

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
Footer Links does not have enough space between them

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #34901


## What is the new behavior?
Added required 8 px space between them

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
Added other small fixes to make it work on small screens less that with 400px and search results when scrolling soes not interfere wit the top navbar items

ajitsinghkaler commented:
@gkalpak can you look into this

mary-poppins commented:
You can preview e52ca73 at https://pr35098-e52ca73.ngbuilds.io/.
You can preview c3e377c at https://pr35098-c3e377c.ngbuilds.io/.

gkalpak commented on aio/src/styles/1-layouts/_footer.scss:
```diff
@@ -69,7 +69,7 @@ footer {
       li {
         list-style-type: none;
-        padding: 0px;
+        padding: 4px 0;
```
Now that the items have more space between them, it is more difficult to distinguish the different groups in narrow screens (where the groups are stacked one under another).

Can you, also update the margin of the group headers [here][1]:

```diff
-    margin: 0 0 16px;
+    margin: 8px 0 12px;
```

[1]: https://github.com/angular/angular/blob/e52ca73f3327ca5a17d82ee121ecfcae872a021a/aio/src/styles/1-layouts/_footer.scss#L43

ajitsinghkaler replied:
Added this

gkalpak commented on aio/src/styles/1-layouts/_marketing-layout.scss:
```diff
@@ -97,7 +97,6 @@ section#intro {
 
   .hero-logo {
     display: flex;
-    width: 400px;
```
Can you post a before/after screenshot, because I don't see any difference with and without this style.

ajitsinghkaler replied:
![Angular](https://user-images.githubusercontent.com/39260684/73775479-f5f9f400-47ab-11ea-9a05-f9a3292354de.png)
Before removing the width:400px

![Angular (1)](https://user-images.githubusercontent.com/39260684/73775562-25106580-47ac-11ea-9653-64bd6e5dceec.png)
After removing the width:400px

gkalpak replied:
I see, thx for the screenshots :+1:

gkalpak commented on aio/src/styles/2-modules/_search-results.scss:
```diff
@@ -6,7 +6,8 @@ aio-search-results {
     overflow: auto;
-    padding: 68px 32px 0;
+    padding: 0px 32px;
+    border-top: 68px solid;
```
Interesting solution :smiley:
Can you please use an explicit border color. The default value, `currentcolor`, only works here because this elements happens to have the same background color and the color defined on `<body>` (`#444`):

```diff
-    border-top: 68px solid;
+    border-top: 68px solid transparent;
```
(We could also use #444, but then we would have to update it in two places if we decided to change the background color.)

ajitsinghkaler replied:
added this

gkalpak reviewed (changes requested):
Nice catches! Thx, @ajitsinghkaler.
Can you also please fix the typos in the commit messages:
- `lighouse` --> `lighthouse`
- `prperly` --> `properly`
- `searh` --> `search`

ajitsinghkaler committed:
fix(docs-infra): size footer links to appropriate size for SEO

Footer links did not have enough space between them, so lighthouse was
reporting that tap targets are not appropriately sized. Added the
required 8px space between links.

Also updated the margin of group headers accordingly.

Fixes #34901

ajitsinghkaler committed:
fix(docs-infra): fix CSS issues on home page and search results

On home page, image size set to 400px, which make the page not render
properly on devices below the size of 400px width.

Also, when search results were scrolled, they interfered with the top
nav items. Added border at the top of the search results container so
that the results are not visible under the nav items during scrolling.

ajitsinghkaler commented:
@gkalpak Made the changes please have a look

mary-poppins commented:
You can preview 1f7e6c0 at https://pr35098-1f7e6c0.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak these same flakey failure can you look into this.

gkalpak commented:
Thx, @ajitsinghkaler.
<sub>FYI: I made some minor changes to the commit messages (mostly fixing typos and missing punctuation).</sub>

mary-poppins commented:
You can preview 74bfad5 at https://pr35098-74bfad5.ngbuilds.io/.

IgorMinar commented:
@ajitsinghkaler thanks for fixing the issue!!!
